extensions-core/lookups-cached-global/src/main/java/org/apache/druid/server/lookup/namespace/cache/CacheScheduler.java

```diff
@@ -253,6 +259,11 @@ private void tryUpdateCache(String currentVersion) throws Exception
           throw t;
         }
       }
+      finally {
```

Comment:
Nit: @rohangarg , do you think it would be cleaner if we return the state of `updatedCacheSuccessfully` from `tryUpdateCache` and use that value to update the future `firstLoadFinishedSuccessfully` in `updateCache`.

I think it would help us avoid updating the future in both `updateCache` and `tryUpdateCache`.

Reply:
done